### PR TITLE
JAMES-3754 Update IMAP ID Extension - support return id fields response when configured

### PR DESCRIPTION
**Pull request by vttranlina**
ref https://www.rfc-editor.org/rfc/rfc2971.html

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

File: server/protocols/protocols-imap4/src/test/resources/imapServerIdCommandResponseFields.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <idleTimeIntervalUnit>SECONDS</idleTimeIntervalUnit>
+    <enableIdle>true</enableIdle>
+    <inMemorySizeLimit>64K</inMemorySizeLimit>
+    <literalSizeLimit>128K</literalSizeLimit>
+    <plainAuthDisallowed>false</plainAuthDisallowed>
+    <gracefulShutdown>false</gracefulShutdown>
+    <idCommandResponse>
+        <field name="name" value="Apache James"/>
+        <field name="version" value="3.9.0"/>
+    </idCommandResponse>
```
Comment (chibenwa) [inline server/protocols/protocols-imap4/src/test/resources/imapServerIdCommandResponseFields.xml:18]: TODO add this into the HELM charts!

**Comment by Arsnael** [comment]
Need a rebase

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
Ci fail
`org.apache.james.backends.opensearch.ClientProviderImplConnectionAuthESIgnoreSSLCheckTest.(?)`
it loosk not related to this pr
It passed on my local
Please re-run CI?

**Comment by Arsnael** [comment]
Opensearch image not started correctly with testcontainers, usual issue. I think I did restart the build same time as you commented :)

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**